var/spack/repos/builtin/packages/py-celery/package.py

```diff
@@ -65,10 +67,29 @@ class PyCelery(PythonPackage):
     depends_on("py-click-repl@0.2.0:", when="@5.2.0:", type=("build", "run"))
     depends_on("py-pytz@2019.3:", type=("build", "run"))
     depends_on("py-pytz@2021.3:", type=("build", "run"), when="@5.2.3")
-    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"))
+    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"), when="@:5.0.0")
     depends_on("py-billiard@3.6.4.0:3", type=("build", "run"), when="@5.2.3")
     depends_on("py-kombu@4.6.11", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-kombu@5.0.0:", when="@5.0.0:5.0", type=("build", "run"))
     depends_on("py-kombu@5.2.3:5", when="@5.2.0:5.2", type=("build", "run"))
     depends_on("py-vine@1.3.0", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-vine@5.0.0:5", when="@5.0.0:5", type=("build", "run"))
+
+    with when("@5.2.3:5"):
+        depends_on("py-cryptography", type=("build", "run"))
+
+    with when("@5.3.4:5"):
```

Comment:
Not a huge fan of `with when("@...")` because I need to merge them all in my head and all the different versions of deps that are/have been required are scattered around the file. I usually structure things like in `py-fiona`:
```
# filename(s) where deps can be found
deps sorted in same order as file
with different lines for changing version reqs

# historical deps
things that are no longer required in the latest version
```
But this is optional, just helps me review.